src/lib/electricity-maps/index.ts

```
@@ -0,0 +1,119 @@
+                        total_carbon_intensity += carbon_intensity.value * hourly_ratios[index];
+                    }
+                ));
+                return {
+                    ...model_param,
+                    carbon_intensity: total_carbon_intensity * power_consumption,
+                    unit: unit
+                }
+            });
+    }
+    initAuth() {
+        dotenv.config();
+        const token = process.env.EMAPS_TOKEN;
+        if (!token) {
+            throw new AuthorizationError(
+                this.errorBuilder({
+                    message: 'Missing token',
+                    scope: 'authorization',
+                })
+            );
+        }
+        this.authorizationHeader = `auth-token: ${token}`;
+    }
+    private async get_carbon_intensity(longitude: number, latitude: number, start: dayjs.Dayjs, end: dayjs.Dayjs): Promise<KeyValuePair[]> {
+        const parameters = {
+            lon: longitude,
+            lat: latitude,
+            start: start,
+            end: end
+        }
+        if (end.diff(start, 'days') > 10) {
+            throw InputValidationError(
+                this.errorBuilder({
+                    message: 'The maximum duration is 10 days for the Emaps API.',
+                    scope: 'input',
+                })
+            )
+        }
+        const url = `${BASE_URL}/carbon-intensity/past-range`;
+        return axios.get(
+            url,
+            {
+                params: parameters,
+                headers: {
+                    'auth-token': this.authorizationHeader,
+                }
+            }
+        ).then(
+            (response)=>{
+                if (response.status !== 200) {
+                    throw new APIRequestError(
+                        this.errorBuilder({
+                            message: `Error: ${JSON.stringify(response.status)}`,
+                        })
+                    );
+                }
```

Do we want to handle more response codes here explicitly?
Like can we use the `AuthorizationError` to throw a Autorization error if the token is invalid and we return a 401 status for example.

---

Yes we could add this but rather in a subsequent PR.